scripts/update_docs_toc.py

```diff
@@ -0,0 +1,79 @@
+# /// script
+# requires-python = ">=3.13"
+# dependencies = ["markdown~=3.8", "mkdocs~=1.6.1"]
+# ///
+
+"""Update `docs/README.md`'s TOC."""
+
+from __future__ import annotations
+
+import re
+from collections.abc import Generator
+from pathlib import Path
+
+from markdown import Markdown
+from mkdocs.structure.toc import AnchorLink, TableOfContents, get_toc
+
+_THIS_FILE = Path(__file__)
+_REPO_ROOT = _THIS_FILE.parent.parent
+_DOCS_README = _REPO_ROOT / "docs" / "README.md"
+
+_TOC_START_MARKER = "<!-- TOC start -->\n\n"
+_TOC_END_MARKER = "\n\n<!-- TOC end -->"
+_TOC_SEPARATOR = " |\n"
+
+_CODE_BLOCK = re.compile(
+    r"""(?xm)
+        ^(```+).*\n
+        (?>(?!\1).*\n)*?
+        \1$
+    """
+)
+
+
+def _iter_links(toc: TableOfContents) -> Generator[AnchorLink]:
+    for link in toc:
+        yield link
+        yield from link.children
+
+
+def _get_new_toc(sanitized_contents: str) -> str:
+    parser = Markdown(extensions=["toc"])
+    parser.convert(sanitized_contents)
+    toc = get_toc(parser.toc_tokens)  # type: ignore  # noqa: PGH003
```

Comment:
Can you add a comment explaining why you are ignoring the type-checker and linter errors here?